sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -477,11 +487,51 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
     }
 
     hdr = ipv6->data;
-    payload = ipv6->next;       /* TODO: parse extension headers */
+    payload = ipv6->next;
 
     if (ng_ipv6_addr_is_multicast(&hdr->dst)) {
         _send_multicast(iface, pkt, ipv6, payload, prep_hdr);
     }
+    else if ((ng_ipv6_addr_is_loopback(&hdr->dst)) ||
+             ((iface == KERNEL_PID_UNDEF) && /* or registered to iface */
+              ((iface = ng_ipv6_netif_find_by_addr(&tmp, &hdr->dst)) != KERNEL_PID_UNDEF)) ||
+             ((iface != KERNEL_PID_UNDEF) &&
+              (ng_ipv6_netif_find_addr(iface, &hdr->dst) != NULL))) {
```

Comment:
Aren't their three conditions? And what or who is registered to which interface?